### PR TITLE
add cabal parser

### PR DESCRIPTION
**Pull request by alunduil**
The Cabal parser is designed similarly to the yarn parser and utilizes a
remote service to parse the Cabal file.

- [x] Have you followed the guidelines for [contributors](http://docs.libraries.io/contributorshandbook)?
- [x] Have you checked to ensure there aren't other open pull requests on the repository for a similar change?
- [x] Is there a corresponding ticket for your pull request?
- [x] Have you written new tests for your changes?
- [ ] Have you successfully run the project with your changes locally?

Relates to #339.

### CODE REVIEW

**Comment by alunduil** [comment]
I'm not sure why the Swift test is currently failing.  Would it be best to fix it here?  Also, feel free to let me know if I got any of the style wrong.

If there was any spam due to my travis failures, I apologize.

**Comment by andrew** [comment]
Looks like the swift parsing service kicked the bucket over the weekend, I've started it back up and the tests appear happier again now 👍

File: lib/bibliothecary/parsers/haskell.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+module Bibliothecary
+  module Parsers
+    class Haskell
+      include Bibliothecary::Analyser
+
+      def self.mapping
+        {
+          /.*\.cabal$/ => {
+            kind: 'lockfile',
```
Comment (andrew) [inline lib/bibliothecary/parsers/haskell.rb]: this should be `'manifest'`, lockfiles are where the full list of transitive dependencies and all resolved versions are stored, which looks to be like `cabal freeze` generates one called `cabal.config`

File: lib/bibliothecary/parsers/haskell.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+require "json"
+
+module Bibliothecary
+  module Parsers
+    class Haskell
```
Comment (andrew) [inline lib/bibliothecary/parsers/haskell.rb]: The naming convention that I've been following has been to name the class the same as the registry name in Libraries.io, which in this case is `Hackage`

**Comment by andrew** [comment]
I made some tweaks and pushed a branch based on yours: https://github.com/alunduil/bibliothecary/compare/add-cabal...librariesio:cabel-tweaks

**Comment by andrew** [comment]
I've also added support for parsing `cabal.config` lockfiles: https://github.com/librariesio/bibliothecary/compare/cabel-tweaks...cabal-config

File: lib/bibliothecary/parsers/haskell.rb (hunk 1 above)
Comment (alunduil) [reply lib/bibliothecary/parsers/haskell.rb]: I wasn't sure about this one.  Thanks for correcting it.

File: lib/bibliothecary/parsers/haskell.rb (hunk 2 above)
Comment (alunduil) [reply lib/bibliothecary/parsers/haskell.rb]: Also wasn't clear.  Thanks again for correcting.

**Comment by alunduil** [comment]
@andrew If you've merged these changes already should we close this pull request or do we still want to merge this?

**Comment by andrew** [comment]
lets keep this open for now, once I've got your cabal-parser service running we'll merge this in and the cabal-config branch and 🚢  a new release

**Comment by andrew** [comment]
@alunduil I've deployed your service to cabal.libraries.io, once https://github.com/librariesio/bibliothecary/pull/413 is merged I'll push a new version with this included

**Pull request merged**